src/main/java/reposense/util/FileUtil.java

```diff
@@ -122,8 +127,67 @@ public static void copyDirectoryFiles(Path src, Path dest) throws IOException {
         }
     }
 
+    /**
+     * Generates a list of paths of {@code fileType} contained in the given {@code directoryName} directory.
+     */
+    public static List<Path> listFilePath(String directoryName, String fileType) {
+        Path directory = Paths.get(directoryName);
+        List<Path> filePaths = null;
+        try {
+            filePaths = Files.walk(directory)
+                    .filter(p -> p.toString().endsWith(fileType))
+                    .distinct()
+                    .collect(Collectors.toList());
+        } catch (IOException ioe) {
+            logger.log(Level.SEVERE, ioe.getMessage(), ioe);
+        }
+
+        return filePaths;
+    }
+
+    /**
+     * Zips all the JSON files contained in the {@code sourcePath} directory.
+     * Creates the zip folder in the {@code outputPath}
+     */
+    public static void zipJson(String outputPath, String sourcePath) {
+        String summaryJson = "summary.json";
+        List<Path> allJsonFiles = listFilePath(outputPath + File.separator + sourcePath, ".json");
+
+        //byte buffer for I/O
+        ByteBuffer buffer = ByteBuffer.allocate(1 << 10);
+        int length;
+        try (FileOutputStream fos = new FileOutputStream(outputPath + File.separator + Constants.JSON_ZIP_FILE);
+             ZipOutputStream zos = new ZipOutputStream(fos)) {
+            for (int i = 0; i < allJsonFiles.size(); i++) {
+                try (FileInputStream fis = new FileInputStream(allJsonFiles.get(i).toFile())) {
+
+                    // begin writing a new ZIP entry, positions the stream to the start of the entry data
+                    if (isFilePathValid(allJsonFiles.get(i), summaryJson)) {
+                        zos.putNextEntry(new ZipEntry(allJsonFiles.get(i).getFileName().toString()));
+                    } else {
+                        zos.putNextEntry(new ZipEntry(allJsonFiles.get(i).getParent().getFileName().toString()
+                                + File.separator + allJsonFiles.get(i).getFileName().toString()));
+                    }
+
+                    while ((length = fis.read(buffer.array())) > 0) {
+                        zos.write(buffer.array(), 0, length);
+                    }
+                }
+            }
+        } catch (IOException ioe) {
+            logger.log(Level.SEVERE, ioe.getMessage(), ioe);
+        }
+    }
+
     private static String attachJsPrefix(String original, String prefix) {
         return "var " + prefix + " = " + original;
     }
 
+    /**
+     * Checks if the given {@code path} is of {@code file}
+     */
+    private static boolean isFilePathValid(Path path, String file) {
```

Comment:
The method name does not reflect what it is doing, should be `checkFileName`? Update the header comments accordingly.